### PR TITLE
fix: throw error so test failures are not swallowed

### PR DESCRIPTION
**Pull request by pauldambra**
So obvious in retrospect

If you listen to the Cypress fail event without re-throwing then you swallow all test failures

"fun"

----

When reviewing https://github.com/PostHog/posthog/pull/17919 I knew the Cypress tests would have to be failing which is what prompted me to check

Introduced in https://github.com/PostHog/posthog/commit/bbb7ed9968e1c495d59da923ee8a85bbf8466f75 (July 10th!)

### CODE REVIEW

**Review by daibhin — APPROVED** [review]

**Comment by neilkakkar** [comment]
nice, thank you!

I just wrote some today and was very confused why the asserts error out but still pass overall. You saved me A TON of digging :D

File: cypress/e2e/signup.cy.ts [hunk 1]
```diff
@@ -74,7 +74,8 @@ describe('Signup', () => {
         cy.get('.Toastify [data-attr="error-toast"]').contains('Inactive social login session.')
     })
 
-    it('Shows redirect notice if redirecting for maintenance', () => {
+    // skip this because it seems to be missing necessary setup feature flag, preflight cloud check...
+    it.skip('Shows redirect notice if redirecting for maintenance', () => {
```
Comment (pauldambra) [inline cypress/e2e/signup.cy.ts:78]: @raquelmsmith can this Cypress test be deleted or should it be fixed?

Comment (raquelmsmith) [reply cypress/e2e/signup.cy.ts:78]: It shouldn't be deleted, we want to make sure this functionality works for when we do use the flag in production. It shouldn't be broken and was working before.. The bit we need for the flag is on line 83 below where we define what is in the decide response.

Comment (pauldambra) [reply cypress/e2e/signup.cy.ts]: Just to check... did the test pass for you running locally before?

The problem we've had is that the tests were passing in CI no matter what. I've had to change the code a little...

* I had to set preflight to cloud in the test itself
* the redirect is hard-coded to use prod US or prod EU URLs so have had to change the redirect code to not redirect from Cypress to production

Comment (raquelmsmith) [reply cypress/e2e/signup.cy.ts]: Yeah it did work locally! Thanks for the fix.

**Pull request merged**